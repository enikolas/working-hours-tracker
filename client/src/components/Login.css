.login-form {
  margin-top: 30px;
}

.login-field {
  min-height: 60px;
}

.login-field input {
  font-size: 19px;
  padding: 11px 14px;
  width: 100%;
}
