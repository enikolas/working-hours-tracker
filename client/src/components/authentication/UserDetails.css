.user-details {
  position: absolute;
  top: 0;
  right: 20px;
  background: #5a717f;
  text-align: center;
  overflow: hidden;
  height: 29px;
  transition: all ease-in 0.5s;
}

.user-details:hover {
  height: 102px;
}

.user-details .employee-name {
  padding: 10px;
  font-weight: bold;
  color: #c5ccd1;
  font-size: 12px;
  min-width: 130px;
}

.user-details button {
  background: transparent;
  color: #fff;
  cursor: pointer;
  opacity: 0.7;
}

.user-details button:hover {
  opacity: 1;
}

.user-details button.logout {
  background-color: #909fa8;
  text-transform: lowercase;
  padding: 2px 6px;
  color: #fff;
  cursor: pointer;
  margin-top: 4px;
}

.user-details button.logout:hover {
  background-color: #c5ccd1;
}

.user-details button.logout:focus {
  outline: 1px solid #f93;
}

.user-details nav li {
  list-style: none;
}
