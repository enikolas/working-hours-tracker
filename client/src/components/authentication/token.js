const API_AUTH_TOKEN = 'achiever-auth-token';

export const setAuthToken = token => localStorage.setItem(API_AUTH_TOKEN, token);

export const getAuthToken = () => localStorage.getItem(API_AUTH_TOKEN);

export const removeAuthToken = () => localStorage.removeItem(API_AUTH_TOKEN);
