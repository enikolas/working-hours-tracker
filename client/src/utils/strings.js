const strings = {
	times: [
		{
			label: 'Entrada',
			cliKeywords: [
				'entrada', 'entrando', 'entrei',
				'chegada', 'chegando', 'cheguei',
				'arrival'
			]
		}, {
			label: 'Ida para almo√ßo',
			cliKeywords: [
				'ida', 'indo',
				'lunch', 'gone-lunch',
				'almoco', 'ida-almoco', 'indo-almoco',
				'fui-almoco', 'partiu-almoco'
			]
		}, {
			label: 'Volta do almo√ßo',
			cliKeywords: [
				'volta', 'voltando', 'voltei',
				'back', 'back-from-lunch',
				'retorno', 'retornando', 'retornei',
				'volta-almoco', 'voltando-almoco', 'voltei-almoco',
				'chegada-almoco', 'chegando-almoco', 'cheguei-almoco'
			]
		}, {
			label: 'Sa√≠da',
			cliKeywords: [
				'saida', 'saindo', 'sai',
				'fui', 'fui-casa', 'partiu-casa',
				'leave', 'left', 'gone-home',
				'ida-casa', 'indo-casa'
			]
		}
	],
	clearCliKeywords: [
		'clear', 'clean', 'reset', 'apaga'
	],
	multipleCliKeywords: [
		'multi', 'multiplos', 'varios'
	],
	sendCallFeedbacks: {
		error: 'Erro ao executar api call para inserir horas:',
		success: 'Api call para inserir horas executada com sucesso!'
	},
	thisTime: 'esta hora',
	hoursLabouredOnThisDay: 'Horas trabalhadas no dia:',
	hoursBalanceOnWeekUpToNow: 'Na semana at√© esse dia',
	hoursBalanceOnWeekUpToNowSurplus: '{0} a mais',
	hoursBalanceOnWeekUpToNowDebt: 'devendo {0}',
	hoursBalanceForToday: 'Balan√ßo de horas atual:',
	storedSuccessfully: 'gravada com sucesso',
	morningPeriod: 'Per√≠odo da manh√£',
	afternoonPeriod: 'Per√≠odo da tarde',
	weeklyCalendar: 'Calend√°rio da semana',
	total: 'Total',
	send: 'Enviar',
	update: 'Atualizar',
	ok: 'OK',
	cancel: 'Cancelar',
	yes: 'Sim',
	no: 'N√£o',
	loading: 'Carregando...',
	noTimesProvided: 'nenhuma hora foi fornecida',
	otherTimesIgnored: 'as outras horas fornecidas foram ignoradas',
	timesFlushed: 'As entradas de tempo foram apagadas',
	cannotInsertDisconnectedTime: ' n√£o pode ser inserida sem o(s) hor√°rio(s) anterior(es)',
	cannotSelectFutureTime: 'Uma data futura n√£o pode ser selecionada',
	markConfirm: 'Quer marcar agora a',
	markNow: 'Marcar agora a',
	timeSentToday: 'As horas de hoje j√° foram enviadas',
	clearConfirm: 'Quer mesmo apagar o arquivo tempor√°rio de horas pra hoje?',
	cliConfirmationAllowed: 'Digite "s" ou "n"',
	cancelled: 'Cancelado',
	todayPage: 'Horas de hoje',
	editPage: 'Editar hor√°rios',
	timeEntryPage: 'Apontamento de horas',
	selectedDate: 'Dia selecionado',
	dateBeingEdited: 'Dia sendo editado',
	todayDate: 'Dia de hoje',
	login: 'Entrada',
	logout: 'Sair',
	logoutConfirm: 'Quer mesmo sair do sistema?',
	username: 'Usu√°rio',
	password: 'Senha',
	vacation: 'F√©rias',
	authenticationError: 'Credenciais inv√°lidas.',
	pageNotFound: 'P√°gina n√£o encontrada',
	pageLoading: 'Carregando...',
	confirmSave: 'Existem altera√ß√µes n√£o salvas. Deseja enviar?',
	invalidTime: 'Hor√°rios inv√°lidos',
	invalidAddTime: 'N√£o √© poss√≠vel marcar presen√ßa em um hor√°rio anterior',
	submitTimeSuccess: 'Horas registradas com sucesso!',
	goBack: 'Voltar',
	helloName: 'Ol√° ',
	warning: 'Aten√ß√£o',
	confirmation: 'Confirma√ß√£o',
	projectPhase: 'Projeto/fase',
	activity: 'Atividade',
	noLunchPause: 'Sem pausa para almo√ßo',
	normalDay: 'Dia normal',
	hourEntryMode: 'Entrada de horas',
	specialDay: 'Dia especial',
	specialDays: {
		medical: 'Dispensa m√©dica',
		birthdayLeave: 'O-tanjoubi'
	},
	success: 'üëç',
	error: 'Erro!',
	passwordChange: 'Mudar senha',
	reportDownload: 'Relat√≥rio mensal',
	oldPassword: 'Senha antiga',
	newPassword: 'Nova senha',
	reportMonth: 'M√™s do relat√≥rio',
	selectMonth: 'selecione o m√™s...',
	insertNewPassword: 'insira a nova senha',
	confirmNewPassword: 'confirme a nova senha',
	passwordChangeSuccessful: 'A senha foi alterada com sucesso!'
};

module.exports = strings;
