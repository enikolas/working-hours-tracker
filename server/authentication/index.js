require('dotenv').config();

const HEADER_REGEX = /Bearer (.*)$/;

exports.authenticate = ({ req }) => {
  const authorization = req.headers.authorization || '';
	const authorizationHeader = HEADER_REGEX.exec(authorization);

	if (!authorizationHeader) {
		return null;
	}

	const token = authorizationHeader[1];

	return token;
};
