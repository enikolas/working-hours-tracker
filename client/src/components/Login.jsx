/* global window */

import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from '@apollo/client/react/hoc';

import * as queries from '../queries';
import AlertModal from './ui/modals/AlertModal';
import strings from '../utils/strings';
import { setAuthToken } from './authentication/token';

import './Login.css';

class Login extends React.Component {
	constructor(props) {
		super(props);

		this.onSubmit = this.onSubmit.bind(this);

		this.state = {
			username: '',
			password: '',
			errorMessage: ''
		};
	}

	onChangeField(field) {
		return (event) => {
			this.setState({ [field]: event.target.value });
		};
	}

	onCloseAlert() {
		return () => {
			this.setState({ errorMessage: '' });
		};
	}

	onSubmit(event) {
		event.preventDefault();

		const { username, password } = this.state;

		this._signIn(username, password);
	}

	_shouldSubmitBeAvailable() {
		return this.state.username && this.state.password;
	}

	async _signIn(username, password) {
		const response = await this.props.signIn({
			variables: {
				user: username,
				password
			}
		});

		if (response) {
			if (response.errors && response.errors.length) {
				this.setState({ errorMessage: response.errors[0].message });
			} else {
				this.setState({ errorMessage: '' });
				const { token } = response.data.signIn;
				setAuthToken(token);
				window.location.reload();
			}
		}
	}

	render() {
		return (
			<React.Fragment>
				<div className="column column-nav" />
				<div className="column column-actions">
					<h2 className="current-date">
						<strong>{strings.login}</strong>
					</h2>
					<main>
						<AlertModal
							title={strings.error}
							active={Boolean(this.state.errorMessage)}
							content={this.state.errorMessage}
							onClose={this.onCloseAlert}
						/>
						<form onSubmit={this.onSubmit} className="login-form">
							<div className="login-field">
								<input
									type="text"
									name="username"
									placeholder={strings.username}
									onChange={this.onChangeField('username')}
								/>
							</div>
							<div className="login-field">
								<input
									type="password"
									name="password"
									placeholder={strings.password}
									onChange={this.onChangeField('password')}
								/>
							</div>
							<button
								type="submit"
								className="send send-login"
								disabled={!this._shouldSubmitBeAvailable()}
							>
								{strings.send}
							</button>
						</form>
					</main>
				</div>
			</React.Fragment>
		);
	}
}

export default graphql(queries.signIn, { name: 'signIn' })(Login);

Login.propTypes = {
	signIn: PropTypes.func
};

Login.defaultProps = {
	signIn: () => {}
};
