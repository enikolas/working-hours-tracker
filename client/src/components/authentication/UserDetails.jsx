import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql } from '@apollo/client/react/hoc';
import { compose } from 'recompose';

import * as queries from '../../queries';
import strings from '../../utils/strings';
import { getAuthToken, removeAuthToken } from './token';
import { clearTodayStorage } from '../../utils';
import UserOptions from '../userOptions/UserOptions';

import './UserDetails.css';

const _logout = (event) => {
	event.preventDefault();
	removeAuthToken();
	clearTodayStorage();
	window.location.reload();
};

class UserDetails extends Component {
	constructor(props) {
		super(props);

		this.state = {
			authenticated: false
		};

		this.openMonthlyReport = this.openMonthlyReport.bind(this);
		this.changePassword = this.changePassword.bind(this);
	}

	componentWillReceiveProps(nextProps) {
		const { userDetails, loading, error } = nextProps.userDetailsQuery;

		if (error || (!loading && !userDetails)) {
			removeAuthToken();
			this.setState({ authenticated: false });
		} else {
			const token = getAuthToken();
			const authenticated = Boolean(token && userDetails.name);
			this.setState({ authenticated });
		}
	}

	/* eslint-disable class-methods-use-this */

	openMonthlyReport(monthData) {
		console.log({ monthData });
	}

	async changePassword(passwordData) {
		const currentPassword = passwordData.old;
		const newPassword = passwordData.new;

		const response = await this.props.changePassword({
			variables: {
				currentPassword,
				newPassword
			}
		});

		if (response) {
			if (response.errors && response.errors.length) {
				// TODO: shows error message
			} else {
				removeAuthToken();
				window.location.reload();
			}
		}
	}

	/* eslint-enable class-methods-use-this */

	render() {
		const { userDetails, loading, error } = this.props.userDetailsQuery;

		if (loading) {
			return <div>Loading...</div>;
		}

		const { authenticated } = this.state;

		if (authenticated && !error) {
			const [firstName] = userDetails.name.split(' ');

			return (
				<div className="user-details">
					<div className="employee-name">
						{`${strings.helloName} ${firstName}`}
					</div>
					<nav>
						<UserOptions
							reportCall={this.openMonthlyReport}
							passwordCall={this.changePassword}
						/>
						<li>
							<button className="logout" onClick={_logout}>
								{strings.logout}
							</button>
						</li>
					</nav>
				</div>
			);
		}
		return null;
	}
}

export default compose(
	graphql(queries.userDetails, { name: 'userDetailsQuery' }),
	graphql(queries.changePassword, { name: 'changePassword' })
)(UserDetails);

UserDetails.propTypes = {
	userDetailsQuery: PropTypes.object,
	changePassword: PropTypes.func
};

UserDetails.defaultProps = {
	userDetailsQuery: {},
	changePassword: () => {}
};
