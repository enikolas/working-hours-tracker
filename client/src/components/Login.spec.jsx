
import React from 'react';
import { shallow } from 'enzyme';

import Login from './Login';

import { getAuthToken } from './authentication/token';
import strings from '../utils/strings';

describe('Login', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<Login signIn={() => {}} />);
		expect(wrapper).toMatchSnapshot();
	});
	describe('_signIn', () => {
		const username = 'teste';
		const password = 'super-secret';
		it('should correctly sign in and redirect user', async (done) => {
			const promiseSignIn = Promise.resolve({ data: { signIn: { token: 'token' } } });
			const signIn = jest.fn(() => promiseSignIn);
			window.location.reload = jest.fn();

			const wrapper = shallow(<Login signIn={signIn} />);
			wrapper.setState({ username, password });
			const form = wrapper.find('form');
			form.simulate('submit', { preventDefault: jest.fn() });

			await promiseSignIn;

			expect(signIn).toHaveBeenCalledWith({ variables: { user: username, password } });
			expect(getAuthToken()).toEqual('token');
			expect(window.location.reload).toHaveBeenCalled();
			done();
		});
		it('should setState to error message if failed', async (done) => {
			const throwWrapper = async () => {
				const promiseSignIn = Promise.reject(new Error('error'));
				const signIn = jest.fn(() => promiseSignIn);
				window.location.reload = jest.fn();

				const wrapper = shallow(<Login signIn={signIn} />);
				wrapper.setState({ username, password });
				const form = wrapper.find('form');
				form.simulate('submit', { preventDefault: jest.fn() });

				await promiseSignIn;

				expect(wrapper.state('errorMessage')).toEqual(strings.authenticationError);
				expect(window.localtion.reload).not.toHaveBeenCalled();
			};
			throwWrapper();
			done();
		});
	});
	describe('onChangeField', () => {
		it('should update if input is changed', () => {
			const wrapper = shallow(<Login signIn={jest.fn()} />);
			const userInput = wrapper.find('[type="text"]');
			userInput.simulate('change', { target: { value: 'Hello' } });
			expect(wrapper.state('username')).toEqual('Hello');
		});
	});
});
