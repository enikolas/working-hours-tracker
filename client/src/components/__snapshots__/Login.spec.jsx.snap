// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    signIn={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="column column-nav"
        />,
        <div
          className="column column-actions"
        >
          <h2
            className="current-date"
          >
            <strong>
              Entrada
            </strong>
          </h2>
          <main>
            <AlertModal
              active={false}
              content=""
              onClose={[Function]}
              title="Erro!"
            />
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="login-field"
              >
                <input
                  name="username"
                  onChange={[Function]}
                  placeholder="Usuário"
                  type="text"
                />
              </div>
              <div
                className="login-field"
              >
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Senha"
                  type="password"
                />
              </div>
              <button
                className="send send-login"
                disabled={true}
                type="submit"
              >
                Enviar
              </button>
            </form>
          </main>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "column column-nav",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="current-date"
            >
              <strong>
                Entrada
              </strong>
            </h2>,
            <main>
              <AlertModal
                active={false}
                content=""
                onClose={[Function]}
                title="Erro!"
              />
              <form
                className="login-form"
                onSubmit={[Function]}
              >
                <div
                  className="login-field"
                >
                  <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Usuário"
                    type="text"
                  />
                </div>
                <div
                  className="login-field"
                >
                  <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Senha"
                    type="password"
                  />
                </div>
                <button
                  className="send send-login"
                  disabled={true}
                  type="submit"
                >
                  Enviar
                </button>
              </form>
            </main>,
          ],
          "className": "column column-actions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Entrada
              </strong>,
              "className": "current-date",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Entrada",
              },
              "ref": null,
              "rendered": "Entrada",
              "type": "strong",
            },
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <AlertModal
                  active={false}
                  content=""
                  onClose={[Function]}
                  title="Erro!"
                />,
                <form
                  className="login-form"
                  onSubmit={[Function]}
                >
                  <div
                    className="login-field"
                  >
                    <input
                      name="username"
                      onChange={[Function]}
                      placeholder="Usuário"
                      type="text"
                    />
                  </div>
                  <div
                    className="login-field"
                  >
                    <input
                      name="password"
                      onChange={[Function]}
                      placeholder="Senha"
                      type="password"
                    />
                  </div>
                  <button
                    className="send send-login"
                    disabled={true}
                    type="submit"
                  >
                    Enviar
                  </button>
                </form>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "active": false,
                  "content": "",
                  "onClose": [Function],
                  "title": "Erro!",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="login-field"
                    >
                      <input
                        name="username"
                        onChange={[Function]}
                        placeholder="Usuário"
                        type="text"
                      />
                    </div>,
                    <div
                      className="login-field"
                    >
                      <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Senha"
                        type="password"
                      />
                    </div>,
                    <button
                      className="send send-login"
                      disabled={true}
                      type="submit"
                    >
                      Enviar
                    </button>,
                  ],
                  "className": "login-form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        name="username"
                        onChange={[Function]}
                        placeholder="Usuário"
                        type="text"
                      />,
                      "className": "login-field",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "username",
                        "onChange": [Function],
                        "placeholder": "Usuário",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Senha"
                        type="password"
                      />,
                      "className": "login-field",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Senha",
                        "type": "password",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Enviar",
                      "className": "send send-login",
                      "disabled": true,
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Enviar",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "main",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="column column-nav"
          />,
          <div
            className="column column-actions"
          >
            <h2
              className="current-date"
            >
              <strong>
                Entrada
              </strong>
            </h2>
            <main>
              <AlertModal
                active={false}
                content=""
                onClose={[Function]}
                title="Erro!"
              />
              <form
                className="login-form"
                onSubmit={[Function]}
              >
                <div
                  className="login-field"
                >
                  <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Usuário"
                    type="text"
                  />
                </div>
                <div
                  className="login-field"
                >
                  <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Senha"
                    type="password"
                  />
                </div>
                <button
                  className="send send-login"
                  disabled={true}
                  type="submit"
                >
                  Enviar
                </button>
              </form>
            </main>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "column column-nav",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="current-date"
              >
                <strong>
                  Entrada
                </strong>
              </h2>,
              <main>
                <AlertModal
                  active={false}
                  content=""
                  onClose={[Function]}
                  title="Erro!"
                />
                <form
                  className="login-form"
                  onSubmit={[Function]}
                >
                  <div
                    className="login-field"
                  >
                    <input
                      name="username"
                      onChange={[Function]}
                      placeholder="Usuário"
                      type="text"
                    />
                  </div>
                  <div
                    className="login-field"
                  >
                    <input
                      name="password"
                      onChange={[Function]}
                      placeholder="Senha"
                      type="password"
                    />
                  </div>
                  <button
                    className="send send-login"
                    disabled={true}
                    type="submit"
                  >
                    Enviar
                  </button>
                </form>
              </main>,
            ],
            "className": "column column-actions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Entrada
                </strong>,
                "className": "current-date",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Entrada",
                },
                "ref": null,
                "rendered": "Entrada",
                "type": "strong",
              },
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <AlertModal
                    active={false}
                    content=""
                    onClose={[Function]}
                    title="Erro!"
                  />,
                  <form
                    className="login-form"
                    onSubmit={[Function]}
                  >
                    <div
                      className="login-field"
                    >
                      <input
                        name="username"
                        onChange={[Function]}
                        placeholder="Usuário"
                        type="text"
                      />
                    </div>
                    <div
                      className="login-field"
                    >
                      <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Senha"
                        type="password"
                      />
                    </div>
                    <button
                      className="send send-login"
                      disabled={true}
                      type="submit"
                    >
                      Enviar
                    </button>
                  </form>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "active": false,
                    "content": "",
                    "onClose": [Function],
                    "title": "Erro!",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="login-field"
                      >
                        <input
                          name="username"
                          onChange={[Function]}
                          placeholder="Usuário"
                          type="text"
                        />
                      </div>,
                      <div
                        className="login-field"
                      >
                        <input
                          name="password"
                          onChange={[Function]}
                          placeholder="Senha"
                          type="password"
                        />
                      </div>,
                      <button
                        className="send send-login"
                        disabled={true}
                        type="submit"
                      >
                        Enviar
                      </button>,
                    ],
                    "className": "login-form",
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <input
                          name="username"
                          onChange={[Function]}
                          placeholder="Usuário"
                          type="text"
                        />,
                        "className": "login-field",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "name": "username",
                          "onChange": [Function],
                          "placeholder": "Usuário",
                          "type": "text",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <input
                          name="password"
                          onChange={[Function]}
                          placeholder="Senha"
                          type="password"
                        />,
                        "className": "login-field",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "name": "password",
                          "onChange": [Function],
                          "placeholder": "Senha",
                          "type": "password",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Enviar",
                        "className": "send send-login",
                        "disabled": true,
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Enviar",
                      "type": "button",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "main",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
